lib/v4lsource/v4lsource.go

```diff
@@ -312,6 +316,7 @@ func (s *V4LSource) dequeueFrame() error {
 	} else {
 		// We probably got old buffers from v4l2, ignore them
 		s.Frames().Error("Got invalid buffer, flags %v", v4l2BufFlagsToStrings(buff.Flags))
+		s.Frames().FailedWriting(frame) // FIXME: needed?
```

Comment:
Probably not needed, I'm not 100% sure about the frameforwarder API :P

Reply:
I added some documenting comments to frameforwarder to clarify the API. The idea is that for every call of `GetFrameForWriting()` there should be a corresponding call of either `FinishedWriting()` or `FailedWriting()`, otherwise the frames will leak and fazantix will segfault. So if a subsequent call of dequeue is guaranteed to produce the same buffer index as this one, we should not call `FailedWriting` – otherwise, we should

Reply:
I think it should not have FailedWriting here. But according to the kernel docs this branch should also never be reached. Either the DQBUF syscall returns an error if streaming breaks or the error flag will be set for transitional corrupt frames like at the start of the stream.

If we start getting buffers without the MAPPED flag set then things have gone horribly wrong

Reply:
Ok so let's just replace that with a panic() and be done with it